utils/port_test.go

```diff
@@ -13,3 +15,80 @@ func Test_GetFreePort(t *testing.T) {
 		t.Fatalf("Expected a port > 0 to be available, got %d", port)
 	}
 }
+
+func Test_FindPortInRange(t *testing.T) {
+	cases := []struct {
+		description string
+		s           string
+		port        int
+		errorMsg    string
+	}{
+		{
+			description: "single value",
+			s:           "6667",
+			port:        6667,
+			errorMsg:    "",
+		},
+		{
+			description: "csv",
+			s:           "6667,6668,6669",
+			port:        6667,
+			errorMsg:    "",
+		},
+		{
+			description: "range",
+			s:           "6668-6669",
+			port:        6668,
+			errorMsg:    "",
+		},
+		{
+			description: "invalid single",
+			s:           "abc",
+			port:        0,
+			errorMsg:    `strconv.Atoi: parsing "abc": invalid syntax`,
+		},
+		{
+			description: "invalid lower range",
+			s:           "abc-123",
+			port:        0,
+			errorMsg:    `strconv.Atoi: parsing "abc": invalid syntax`,
+		},
+		{
+			description: "invalid upper range",
+			s:           "123-abc",
+			port:        0,
+			errorMsg:    `strconv.Atoi: parsing "abc": invalid syntax`,
+		},
+		// Seems appvoyer's container lets you map to any port
+		// not sure how to test this case.  Leaving for review
+		// (passes locally)
+		// {
+		// 	description: "unusable ports",
+		// 	s:           "123-124",
+		// 	port:        0,
+		// 	errorMsg:    "all passed ports can not be used",
+		// },
+	}
+
+	for _, c := range cases {
+		t.Run(c.description, func(t *testing.T) {
```

Comment:
I like this!

Reply:
Yeah, been on a big subtest kick since they released it. Its great for unit testing on coveralls.